velox/functions/prestosql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -3109,3 +3109,169 @@ TEST_F(DateTimeFunctionsTest, timeZoneMinute) {
       timezone_minute("2023-", "Pacific/Chatham"),
       "Unable to parse timestamp value: \"2023-\", expected format is (YYYY-MM-DD HH:MM:SS[.MS])");
 }
+
+TEST_F(DateTimeFunctionsTest, toISO8601TestVarchar) {
+  const auto toISO8601 = [&](std::optional<Date> dateObj) {
+    return evaluateOnce<std::string, Date>("to_iso8601(c0)", dateObj);
+  };
+
+  // Date() input: days since epoch
+
+  // Date(0) is 1970-01-01.
+  EXPECT_EQ("1970-01-01T00:00:00.000", toISO8601(Date()));
```

Comment:
It would be better to take a string representation of the Date, construct a Date object out of it and then invoke toISO8601 to check the printed value

auto parseDate = [](const std::string& dateStr) {
    Date returnDate;
    parseTo(dateStr, returnDate);
    return returnDate;
  };

  // Epoch.
  EXPECT_EQ(toISO8601(parseDate("1970-01-01")),"1970-01-01T00:00:00.000");

Reply:
Fair point, fixed the test as such.